src/cli/self_update/shell.rs

```diff
@@ -201,6 +206,46 @@ impl UnixShell for Zsh {
     }
 }
 
+struct Fish;
+
+impl UnixShell for Fish {
+    fn does_exist(&self) -> bool {
+        // fish has to either be the shell or be callable for fish setup.
+        matches!(process().var("SHELL"), Ok(sh) if sh.contains("fish"))
+            || matches!(utils::find_cmd(&["fish"]), Some(_))
+    }
+
+    // > "$XDG_CONFIG_HOME/fish/conf.d" (or "~/.config/fish/conf.d" if that variable is unset) for the user
+    // from <https://github.com/fish-shell/fish-shell/issues/3170#issuecomment-228311857>
+    fn rcfiles(&self) -> Vec<PathBuf> {
+        let p0 = process().var("XDG_CONFIG_HOME").ok().map(|p| {
+            let mut path = PathBuf::from(p);
+            path.push("fish/config.d/rustup.fish");
```

Comment:
This is a typo, as the comment suggests, this should be `conf.d`, not `config.d`